cobigen/cobigen-tsplugin/src/main/java/com/devonfw/cobigen/tsplugin/merger/TypeScriptMerger.java

```diff
@@ -63,107 +86,125 @@ public String getType() {
+     * @param os
+     *            {@link OutputStream} that will be opened for sending data
+     * @param osw
+     *            {@link OutputStreamWriter} used for writing the data to be sent
+     * @throws IOException
+     *             when connection to the server failed
+     * @throws JsonGenerationException
+     *             When generating the JSON from the transfer object failed
+     * @throws JsonMappingException
+     *             When mapping the JSON from the transfer object failed
+     */
+    private void sendRequest(Object dataTO, HttpURLConnection conn, OutputStream os, OutputStreamWriter osw)
+        throws IOException, JsonGenerationException, JsonMappingException {
+        ObjectWriter objWriter;
+        objWriter = new ObjectMapper().writer().withDefaultPrettyPrinter();
+        String jsonMergerTO = objWriter.writeValueAsString(dataTO);
+
+        // We need to escape new lines because otherwise our JSON gets corrupted
+        jsonMergerTO = jsonMergerTO.replace("\\n", "\\\\n");
+
+        osw.write(jsonMergerTO);
+        osw.flush();
+        os.close();
+        conn.connect();
+
+        if (conn.getResponseCode() != HttpURLConnection.HTTP_CREATED) {
+            throw new RuntimeException("Failed : HTTP error code : " + conn.getResponseCode());
         }
     }
 
     /**
      * Reads the output.ts temporary file to get the merged contents
-     * @param base
-     *            base file just for exception handling
-     * @param mergedContents
-     *            merged typescript code
+     * @param importsAndExports
+     *            The part of the code where imports and exports are declared
+     * @param body
+     *            the rest of the body of the source file
      * @return merged contents already beautified
      */
-    private String runBeautifierExcludingImports(File base, String mergedContents) {
-        StringBuilder importsAndExports = new StringBuilder();
-        StringBuilder body = new StringBuilder();
+    private String runBeautifierExcludingImports(String importsAndExports, String body) {
 
-        try (StringReader inR = new StringReader(mergedContents); BufferedReader br = new BufferedReader(inR)) {
-            String line;
-            while ((line = br.readLine()) != null) {
-                if (line.startsWith("import ") || isExportStatement(line)) {
-                    importsAndExports.append(line);
-                    importsAndExports.append(LINE_SEP);
-                } else {
-                    body.append(line);
-                    body.append(LINE_SEP);
-                }
-            }
-        } catch (IOException e) {
-            throw new MergeException(base, "Could not process merged contents for formatting.", e);
-        }
+        FileTO fileTO = new FileTO(body);
+        HttpURLConnection conn = request.getConnection("POST", "Content-Type", "application/json", "beautify");
```

Comment:
Isn't this conn a closable as well? I am not quite sure. If so, we have an open handle here which is not closed.

Reply:
2. is "beautify" the URL? If so, we should start wit naming conventions to not have any name clashes later on. So here it could be tsplugin/beautify

Reply:
`The resource type HttpURLConnection does not implement java.lang.AutoCloseable` 

This is something you already suggested me on a previous pull request 😄. Unfortunately, it is not closeable.